packages/schematics/angular/library/index.ts

```diff
@@ -96,39 +89,70 @@ function updateTsConfig(packageName: string, distRoot: string) {
 function addDependenciesToPackageJson() {
 
   return (host: Tree) => {
-    if (!host.exists('package.json')) { return host; }
-
-    return updateJsonFile(host, 'package.json', (json: PackageJsonPartialType) => {
-
-
-      if (!json['dependencies']) {
-        json['dependencies'] = {};
-      }
-
-      json.dependencies = {
-        '@angular/common': latestVersions.Angular,
-        '@angular/core': latestVersions.Angular,
-        '@angular/compiler': latestVersions.Angular,
-        // De-structure last keeps existing user dependencies.
-        ...json.dependencies,
-      };
-
-      if (!json['devDependencies']) {
-        json['devDependencies'] = {};
-      }
+    const ngCommonDep: NodeDependency = {
+      type: NodeDependencyType.Default,
+      packageName: '@angular/common',
+      version: latestVersions.Angular,
+    };
+    const ngCoreDep: NodeDependency = {
+      type: NodeDependencyType.Default,
+      packageName: '@angular/core',
+      version: latestVersions.Angular,
+    };
+    const ngCompilerDep: NodeDependency = {
+      type: NodeDependencyType.Default,
+      packageName: '@angular/compiler',
+      version: latestVersions.Angular,
+    };
+    const ngCompilerCliDep: NodeDependency = {
+      type: NodeDependencyType.Dev,
+      packageName: '@angular/compiler-cli',
+      version: latestVersions.Angular,
+    };
+    const buildNgPackagr: NodeDependency = {
+      type: NodeDependencyType.Dev,
+      packageName: '@angular-devkit/build-ng-packagr',
+      version: latestVersions.DevkitBuildNgPackagr,
+    };
+    const buildAngular: NodeDependency = {
+      type: NodeDependencyType.Dev,
+      packageName: '@angular-devkit/build-angular',
+      version: latestVersions.DevkitBuildNgPackagr,
+    };
+    const ngPackagr: NodeDependency = {
+      type: NodeDependencyType.Dev,
+      packageName: 'ng-packagr',
+      version: '^3.0.0-rc.2',
+    };
+    const tsickle: NodeDependency = {
+      type: NodeDependencyType.Dev,
+      packageName: 'tsickle',
+      version: '>=0.25.5',
+    };
+    const tslib: NodeDependency = {
+      type: NodeDependencyType.Dev,
+      packageName: 'tslib',
+      version: '^1.7.1',
+    };
+    const typescript: NodeDependency = {
+      type: NodeDependencyType.Dev,
+      packageName: 'typescript',
+      version: latestVersions.TypeScript,
+    };
+    addPackageJsonDependencies(host, [
+      ngCommonDep,
```

Comment:
All these variables are only used once.  An array of object literals would probably be cleaner and less error-prone (it would be easy to forget to add one to the call).